File: 2015/05_Doesnt_He_Have_Intern-Elves_For_This/part_1/main.kt
/**
 * Author: happ_2h https://github.com/happ-2h
 *
 * Solution for Advent of Code 2015 Day 5.1
 */

import java.io.File

fun main(args: Array<String>) {
  var niceCtr = 0

  File("../input.txt").readLines().forEach {
    // At least three vowels
    val vowels = Regex("[aeiou]").findAll(it).count() >= 3

    // 1 letter that appears twice in a row
    val repeated = "([a-z])\\1".toRegex().find(it)?.groups

    // Does not contain the strings ab, cd, pq, or xy
    val noContain = "(ab|cd|pq|xy)".toRegex().find(it)?.groups

    if (vowels == true && repeated != null && noContain == null) ++niceCtr
  }

  println(niceCtr)
}

File: 2015/05_Doesnt_He_Have_Intern-Elves_For_This/part_1/main.js
/**
 * Author: happ_2h https://github.com/happ-2h
 *
 * Solution for Advent of Code 2015 Day 5.1
 */

let input = null;

try {
  input = require('fs').readFileSync("../input.txt", "utf-8")
    .trim().replace(/\r\n|\n|\r/gm, " ").split(" "); // Separate values
} catch (err) {
  console.error(err);
}

if (input !== null) {
  let niceCtr = 0;
  for (const str of input) {
    // At least 3 Vowels
    const vowels = str.match(/[aeiou]/g)?.length >= 3; // Get all vowels then check length

    // 1 letter that appears twice in a row
    const repeated = /([a-z])\1/g.test(str); // Match any character and that same character after

    // Does not contain the strings ab, cd, pq, or xy
    const noContain = !(/(ab|cd|pq|xy)/g.test(str)); // Get the sequence, then logically not the result

    // Combine all results
    if (vowels && repeated && noContain) ++niceCtr;
  }

  console.log(niceCtr);
}
else console.error("Failed to read input.txt");